crates/lang/codegen/src/generator/dispatch.rs

```diff
@@ -523,46 +482,27 @@ impl Dispatch<'_> {
         quote_spanned!(span=>
             const _: () = {
                 #[allow(non_camel_case_types)]
-                pub enum __ink_ConstructorDecoder {
-                    #( #constructors_variants ),*
-                }
+                pub struct __ink_ConstructorDecoder;
 
-                impl ::ink_lang::reflect::DecodeDispatch for __ink_ConstructorDecoder {
-                    fn decode_dispatch<I>(input: &mut I)
-                        -> ::core::result::Result<Self, ::ink_lang::reflect::DispatchError>
-                    where
-                        I: ::scale::Input,
-                    {
-                        match <[::core::primitive::u8; 4usize] as ::scale::Decode>::decode(input)
+                impl ::ink_lang::reflect::ExecuteDispatchable for __ink_ConstructorDecoder {
+                    fn execute_dispatchable() -> ::core::result::Result<(), ::ink_lang::reflect::DispatchError> {
+                        const CAPACITY: usize = 1 << 14;
+                        let mut local_buffer: [u8; CAPACITY] = [0; CAPACITY];
```

Comment:
Why not use the already existing static buffer that each ink! smart contract already has? I do not see a technical reason why using the already existing static buffer should be less efficient than using a local one.

Reply:
We should use a static buffer because it allows us to not allocate the memory in runtime -> less gas.

I will try to rework it to use a static buffer. The problem is that `off_chain` and `off_chain_experemental` don't use a static buffer. So I can't return `&[u8]`. But if you are okey with that idea I will add a static buffer to that engines.

The local buffer is used to fastly create that PR and check the impact=)